src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -10,6 +11,12 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
+        public static IEnumerable<object[]> Encodings_TestData()
```

Comment:
So in essence, the file contains:
- Invalid tests: have test data and test code
- Valid positive tests: have only test code, not data

Maybe we split the file up then?